src/uu/rm/src/rm.rs

```diff
@@ -586,6 +618,32 @@ fn handle_writable_directory(path: &Path, options: &Options, metadata: &Metadata
     }
 }
 
+/// Removes trailing slashes, for example 'd/../////' yield 'd/../' required to fix rm-r4 GNU test
+fn clean_trailing_slashes(path: &Path) -> &Path {
+    let path_str = path.to_str();
+    #[cfg(not(windows))]
+    let directory_separator = b'/';
+    #[cfg(windows)]
+    let directory_separator = b'\\';
+    if let Some(path_str) = path_str {
+        let path_bytes = path_str.as_bytes();
+        let mut idx = path_str.len() - 1;
+        // Checks if element at the end is a '/'
+        if path_bytes[idx] == directory_separator {
+            for i in (0..path_bytes.len()).rev() {
+                // Will break at the start of the continuous sequence of '/', eg: "abc//////" , will break at
+                // "abc/"
+                if path_bytes[i - 1] != directory_separator {
```

Comment:
In iteration `i == 0`, it would try to access index `-1` of the byte slice, which would crash. This bug can be triggered by providing the value `/////`:

```console
$ cargo run -q rm -rf ///////
thread 'main' panicked at src/uu/rm/src/rm.rs:629:31:
attempt to subtract with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[$? = 101]
```

Let's not do that.

Note that simply changing it to `(1..path_bytes.len())` introduces a different edge case in the case of the string `"//////"`.

Please add a test for that. Please remember that tests may be run as root, so doing `rm -rf /` may have bad consequences. Please use the precautions from the `no-preserve-root` test to make sure we don't accidentally nuke developer machines.

Btw, thank you for handling this edge case; this is something that the other PR is lacking.

Reply:
I added a test for the edge case you mentioned. It seems from our existing code, we don't even handle the case when path is root, we end up just printing out to stderr. I changed the error message to resemble GNUs.